python/hail/expr/expressions/base_expression.py

```diff
@@ -625,6 +625,9 @@ def _to_table(self, name):
                         m = source.select_rows(**{name: self})
                     to_return = m.rows().select_globals()
                 else:
+                    if source.col_key is not None:
+                        warn("Column expression sorted by 'col_key'. "
```

Comment:
this is going to be reeeeealy noisy

Reply:
Could we only warn the first time it happens?

Reply:
Have we a mechanism for doing this? Should I add state to the Hail context?